mp/src/game/shared/neo/neo_predicted_viewmodel.cpp

```diff
@@ -429,6 +431,24 @@ void CNEOPredictedViewModel::CalcViewModelView(CBasePlayer *pOwner,
 }
 
 #ifdef CLIENT_DLL
+void CNEOPredictedViewModel::ProcessMuzzleFlashEvent()
+{
+	Vector vAttachment;
+	QAngle dummyAngles;
+	GetAttachment(2, vAttachment, dummyAngles);
```

Comment:
Why the attachment number is `2`? https://developer.valvesoftware.com/wiki/Muzzle_Flash_Lighting recommends using `1`, and in my opinion it looks better:

`2`:
![Screenshot_20240913_133940](https://github.com/user-attachments/assets/306651ea-3a04-412b-b198-a7a9f2d2fd85)

`1`:
![Screenshot_20240913_134236](https://github.com/user-attachments/assets/92bed18d-cace-40d6-98ba-20461ba1d68f)

Reply:
The viewmodel is actually a very big model that extends far in front of you. When close to and facing a wall, the muzzle attachment point will actually be on the other side of the wall and you will not see the environment light at all. Attachment point two is a bit closer to the player and testing a bit in the original I think the original also uses attachment point 2